sentry/src/main/java/io/sentry/clientreport/AtomicClientReportStorage.java

```diff
@@ -0,0 +1,76 @@
+package io.sentry.clientreport;
+
+import io.sentry.ILogger;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+final class AtomicClientReportStorage implements ClientReportStorage {
+
+  private final @NotNull Map<ClientReportKey, AtomicLong> lostEventCounts;
+
+  public AtomicClientReportStorage() {
+    Map<ClientReportKey, AtomicLong> map = new ConcurrentHashMap<>();
+
+    for (DiscardReason discardReason : DiscardReason.values()) {
+      for (DataCategory category : DataCategory.values()) {
+        map.put(
+            new ClientReportKey(discardReason.getReason(), category.getCategory()),
+            new AtomicLong(0));
+      }
+    }
+
+    lostEventCounts = Collections.unmodifiableMap(map);
```

Comment:
Added this as a safety net in case someone changes the implementation and tries to modify entries in the map which would most likely be non thread safe unless synchronized. Some day we're gonna move to a higher API level that also supports the atomic modifications on `ConcurrentHashMap`.

Reply:
Not against it, the downside is that if somebody tries to modify it, it'd throw, we need to be sure that if we do that by mistake, there's a `catch` somewhere handling such a case, we need to degrade gracefully.

Reply:
`catch` blocks are in place in `ClientReportRecorderImpl`. In case something goes wrong it logs but doesn't throw out.